src/HttpLib/HttpRequest.java

```diff
@@ -1,39 +1,50 @@
 package HttpLib;
 
-import HttpLib.Exceptions.HttpFormatException;
+import java.net.URL;
 
 public class HttpRequest {
 
-    // TODO: Create URL class
-    String url;
+    // TODO: Create URL class -> Using Java.net.URL instead -> ON HOLD
+
+    HttpMessageUrl url;
     HttpRequestMethod requestMethod;
-    HttpMessageHeader messageHeader = new HttpMessageHeader();
-    HttpRequestBody body = new HttpRequestBody("");
+    HttpMessageHeader messageHeader;
+    HttpRequestBody body;
 
     // TODO: IsValid method
     // TODO: Getters
 
-    public HttpRequest(String url, HttpRequestMethod requestMethod, HttpMessageHeader messageHeader, HttpRequestBody body) {
-        this.url = url;
+    public HttpRequest(URL url, HttpRequestMethod requestMethod, HttpMessageHeader messageHeader, HttpRequestBody body) {
+        this.url = new HttpMessageUrl(url);
         this.requestMethod = requestMethod;
         this.messageHeader = messageHeader;
         this.body = body;
     }
 
 
     public String toString() {
-        // TODO: todo
-        return "";
+        String request = requestMethod.toString() + url.getFileAndQuery() + "HTTP/1.0\r\n"
+                        + messageHeader.toString()
+                        + body.getLengthString()
+                        + "\r\n"
+                        + body;
+        return request;
     }
 
     public boolean isValid(){
-        // TODO: all composite objects should be valid
+        //TODO: other validation?
+        if (requestMethod.equals(HttpRequestMethod.POST)) {
+            return body != null;
+        }
+        // Headers have pre-check and do not require another check prior to sending
+        // Url has pre-check and does not require another check prior to sending
+
```

Comment:
The if check is wrong.. the request can be of any method type (GET, POST, UPDATE, DELETE, etc..)
We should simply check that every component is present and themselves valid.